views/partials/content-search-results.njk

```diff
@@ -3,7 +3,7 @@
   <li class="mt-10">
     <div>
       <h3 class="text-2xl font-semibold">
-        <a class="text-primary" href="/{{ page.slug }}">{{ page.title }}</a>
+        <a class="text-primary" href="/{{ page.slug }}">{{ page.title  | replace('&amp;', '&')}}</a>
```

Comment:
I'm not sure if this is a good approach for this issue. Maybe you can try `{{ page.title | safe }}`?

Reply:
I made changes as you suggested. I replaced `replace(...` with `| safe`. All is functioning well, both on particular blog page or blog search result page.